modules/delegation/providers/DelegationFormContext.tsx

```diff
@@ -116,17 +118,36 @@ const useDelegationFormActions = (
 //
 // Data provider
 //
-export const DelegationFormProvider: FC<{ mode: DelegationFormMode }> = ({
+export type DelegationFormProviderProps = {
+  mode: DelegationFormMode
+  presetDelegateAddress?: string
+}
+
+export const DelegationFormProvider: FC<DelegationFormProviderProps> = ({
   children,
   mode,
+  presetDelegateAddress,
 }) => {
   const networkData = useDelegationFormNetworkData()
 
   const formObject = useForm<DelegationFormInput>({
-    defaultValues: { delegateAddress: null },
+    defaultValues: { delegateAddress: '' },
     mode: 'onChange',
   })
 
+  useEffect(() => {
+    const currentValue = formObject.getValues('delegateAddress')
+    if (
+      presetDelegateAddress &&
+      isValidAddress(presetDelegateAddress) &&
+      currentValue?.toLowerCase() !== presetDelegateAddress.toLowerCase()
+    ) {
+      console.log('setting default address', presetDelegateAddress)
```

Comment:
Is that intended?

Reply:
No, of course not, thanks for pointing out